### DISCUSSION TITLE
questions, ideas - display and image base64 in the chat/playground

### DISCUSSION THREAD

**Discussion by michelgirault** [langflow-ai/langflow#3956] [General]
Hello everyone,

I'm still new to and I'm just interested to test and try
I have been testing around to use langflow to generate based on input prompt an image from an external api (my own stable diffuision)

Right now I can input, generate an image in base64 and output the text, i was wondering if i c could display the image in the playground or chat? any idea or suggestions?

![Screenshot from 2024-09-28 09-51-49](https://github.com/user-attachments/assets/f7190e8c-e16a-4bfb-8788-123bd0a248f3)
![Screenshot from 2024-09-28 09-52-17](https://github.com/user-attachments/assets/e9ac3c01-4131-41aa-a07e-23e8df37b2d4)

I use a custom components to call and generate the image and output the image in base64

**Comment by michelgirault** [comment]
Any ideas :)

**Reply by BaoMinhChu** [reply]
Reading the doc reveal a Message object that can hold file in "file" parameters. Haven't try it but you can connect this Message to Chat Output to view the image in playground maybe?

**Reply by michelgirault** [reply]
sorry i should have close this topic as i resolved the issue months ago

**Reply by jobs-git** [reply]
how did you resolve it? at least provide the solution

**Comment by michelgirault** [comment]
i shared in discord my solution, anyway this is the extact of my custom component

 # Save the image
        with open(full_path, "wb") as image_file:
            image_file.write(base64.b64decode(image_data))
        #display in chat the image output / link
        image_to_chat = Message(
            text=full_url_path,
            sender="image_gen",
            sender_name="Image",
            content_blocks=[
            ContentBlock(
            title="Media Block",
            contents=[
                MediaContent(type="media", urls=[full_url_path])
                    ]
                )
            ],
        )
